### ISSUE TITLE
Null coalescing of 'default' literal with non-nullable value

### ISSUE THREAD

**Issue by akarpov89** [dotnet/roslyn#21582]
**Version Used**: 
Visual Studio 2017 15.3
**Steps to Reproduce**:
Compile the following code:
```csharp
class C
{
  int M() => default ?? 42;
}
```

**Expected Behavior**:
Compiler raises an error:
error CS0019: Operator '??' cannot be applied to operands of type 'default' and 'int'

**Actual Behavior**:
Code compiles without error or warning and emits code that is equivalent to:
```csharp
class C
{
  int M() => 42;
}
```

This issue is follow-up of the issue #19848.
Somehow I failed to notice this case last time.
I see that case `default ?? 1` is covered [here](https://github.com/dotnet/roslyn/blob/d4dab355b96955aca5b4b0ebf6282575fad78ba8/src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs#L923), but it seems to me that it was simply overlooked as it presented without comment `// ok`.
I'm not entirely sure, because can't find detailed notes, but in my opinion this should be fixed.

/cc @jcouv What do you think?

**Comment by gafter** [comment]
This is not a bug. The expression `default` is not legal where it does not have a target type (i.e. is subject to a conversion-from-expression). In this context it is not legal.

**Comment by gafter** [comment]
Oh, I see the problem is that the compiler accepts this rather than rejects it.

**Comment by jcouv** [comment]
Fixed by #21757 and documented as a compat break. Thanks
